server/routes/referrals/referralsController.ts

```diff
@@ -468,12 +514,16 @@ export default class ReferralsController {
         res.status(400)
         formError = data.error
       } else {
-        await this.interventionsService.setDesiredOutcomesForServiceCategory(accessToken, referralId, {
-          serviceCategoryId,
-          ...data.paramsForUpdate,
-        })
-
-        return res.redirect(`/referrals/${referralId}/form`)
+        try {
+          await this.interventionsService.setDesiredOutcomesForServiceCategory(accessToken, referralId, {
+            serviceCategoryId,
+            ...data.paramsForUpdate,
+          })
+
+          return res.redirect(`/referrals/${referralId}/form`)
+        } catch (e) {
+          formError = createFormValidationErrorOrRethrow(e)
```

Comment:
what is the impact of this? were we not handling formerrors for updating desired outcomes before?

Reply:
I'm really confused about what this method is doing. Andrew quite rightly noted that we weren't using it everywhere, so I thought I'd add it in. I thought it was about surfacing Interventions Service-level errors to the user in a more helpful way, but there seems to be no difference with or without it when testing locally (albeit with stubbed responses, I might not be stubbing them correctly)